api/server.go

```diff
@@ -0,0 +1,189 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost-load-test-ng/config"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/control"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/control/simplecontroller"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/store/memstore"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/user/userentity"
+)
+
+// remove once https://github.com/mattermost/mattermost-load-test-ng/pull/87 is merged
+type MockStatus struct {
+	State           loadtest.State // State of the the load test.
+	NumUsers        int            // Number of active users.
+	NumUsersAdded   int            // Number of users added since the start of the test.
+	NumUsersRemoved int            // Number of users removed since the start of the test.
+	NumErrors       int32          // Number of errors that have occurred.
+	StartTime       time.Time      // Time when the load test was started. This only logs the time when the load test was first started, and does not get reset if it was subsequently restarted.
+}
+
+type API struct {
+	agents map[string]*loadtest.LoadTester
+}
+
+func writeJsonResponse(w http.ResponseWriter, data interface{}) {
+	w.Header().Set("Content-Type", "application/json")
+	_ = json.NewEncoder(w).Encode(data)
+}
+
+func (a *API) createLoadTestHandler(w http.ResponseWriter, r *http.Request) {
+	var config config.LoadTestConfig
+	err := json.NewDecoder(r.Body).Decode(&config)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	u, err := uuid.NewV4()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	newSimpleController := func(id int, status chan<- control.UserStatus) control.UserController {
+		ueConfig := userentity.Config{
+			ServerURL:    config.ConnectionConfiguration.ServerURL,
+			WebSocketURL: config.ConnectionConfiguration.WebSocketURL,
+		}
+		ue := userentity.New(memstore.New(), ueConfig)
+		return simplecontroller.New(id, ue, status)
+	}
+
+	lt := loadtest.New(&config, newSimpleController)
+	a.agents[u.String()] = lt
+
+	writeJsonResponse(w, map[string]string{"loadTestId": u.String()})
```

Comment:
I would try to include the `LoadTester` status in the response every time we change it with the exception of the destroy call since that's final. This way the user of this API doesn't need to make an extra call.